### PR TITLE
Studio V6 Integration- updating  Dirty State and saving files

### PR DESCRIPTION
**Pull request by paulr34**
Updating Dirty State and sending the status to Studio V6 via Post Message

### CODE REVIEW

File: src-shared/rest-api.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const uri = {
   deletePackageNotification: 'deletePackageNotification',
   duplicateEndpoint: '/duplicateEndpoint',
   duplicateEndpointType: '/duplicateEndpointType',
-  dirtyFlag: '/dirtyFlag',
```
Comment (tecimovic) [inline src-shared/rest-api.js:37]: Why has the previous API gotten removed?

File: src/store/zap/actions.js [hunk 2]
```diff
@@ -833,8 +833,10 @@ export function updateSelectedUcComponentState(context, projectInfo) {
   })
 }
 
-export function setDirtyState(context, isDirty) {
-  context.commit('setDirtyState', isDirty)
+export function setDirtyState(context) {
```
Comment (tecimovic) [inline src/store/zap/actions.js:836]: This makes little sense.... Why are we putting dirty state into the store?

**Review by tecimovic — COMMENTED** [review]
I don't understand this change. What are we trying to achieve?

File: src/store/zap/actions.js (hunk 2 above)
Comment (paulr34) [reply src/store/zap/actions.js:836]: this was previously written code that sends the dirty state value in the store to Studio V6

the value was never being updated 

this PR adds a REST_API which updates it

File: src-shared/rest-api.js (hunk 1 above)
Comment (paulr34) [reply src-shared/rest-api.js:37]: adding back. I thought it was not used but it is in the studio-rest-api files..

**Comment by paulr34** [comment]
> I don't understand this change. What are we trying to achieve?

Previously, we adding Studio V6 integration code which sends a postMessage to the parent DOM, Studio V6. This message tells Studio the state of the ZAP file. There was a bug where the state was never changing because we were using websocket incorrectly.

**Pull request closed without merging**